console/v1/types_console_quick_start.go

```diff
@@ -0,0 +1,126 @@
+package v1
+
+import (
+	authorizationv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleQuickStart is an extension for guiding user through various
+// workflows in the OpenShift web console.
+type ConsoleQuickStart struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ConsoleQuickStartSpec `json:"spec"`
+}
+
+// ConsoleQuickStartSpec is the desired quick start configuration.
+type ConsoleQuickStartSpec struct {
+	// displayName is the display name of the Quick Start.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	DisplayName string `json:"displayName"`
+	// icon is a base64 encoded image that are used as Quick Start's
+	// +optional
+	Icon string `json:"icon,omitempty"`
+	// tags is an array of strings that describe the Quick Start.
```

Comment:
The term `array` and `list` are used interchangeably in the descriptions. Choose one for consistency.